src/results.cpp

```diff
@@ -215,7 +215,9 @@ util::Optional<RowExpr> Results::first()
         case Mode::Query:
         case Mode::TableView:
             update_tableview();
-            return m_table_view.size() == 0 ? util::none : util::make_optional(m_table_view.front());
+            return (m_table_view.size() == 0) ||
+                (m_update_policy == UpdatePolicy::Never && !m_table_view.is_row_attached(0)) ?
+                util::none : util::make_optional(m_table_view.front());
```

Comment:
maybe it makes more sense to return a `util::make_optional(RowExpr())` instead? so it returns `none` for empty snapshot, but empty Row for deleted rows?